src/validator.ts

```diff
@@ -303,7 +310,15 @@ function _defaultValidations(): ValidationFunction[] {
       return false;
     }
 
-    function _assertSignaturesMatch(expected: spec.Method, actual: spec.Method, label: string, action: string) {
+    function _assertSignaturesMatch(
+      expected: spec.Method,
+      actual: spec.Method,
+      label: string,
+      action: string,
+      opts: {
+        allowCovariance: boolean;
```

Comment:
Return type covariance.

By the way, does that mean we also have argument type contravariance? Or no?

Reply:
It's only return types and readonly properties. Argument type covariance is also still not allowed.